src/Mafia/Cabal/Dependencies.hs

```diff
@@ -293,9 +304,18 @@ pPackageChange =
 -- TODO instead of `Text -> Maybe a` so we didn't need these two clunky
 -- TODO wrappers below:
 
-pPackageId :: (Char -> Bool) -> Parser PackageId
-pPackageId p = do
-  txt <- A.takeTill p
+pPackageId :: Parser PackageId
+pPackageId = do
+  let
+    isPkgIdChar c
+        | c >= 'a' && c <= 'z' = True
+        | c >= 'A' && c <= 'Z' = True
+        | c >= '0' && c <= '9' = True
+        | c == '-' = True
+        | c == '.' = True
+        | otherwise = False
```

Comment:
Curious where you found these rules? (just curious)

Reply:
Its from memory. I read it in the cabal sources at some point. At the moment google can only find : https://github.com/commercialhaskell/stack/blob/master/src/Stack/Types/PackageName.hs#L76-L84